java/arcs/android/devtools/DevWebServerImpl.kt

```diff
@@ -14,35 +14,63 @@ package arcs.android.devtools
 import arcs.core.util.TaggedLog
 import fi.iki.elonen.NanoHTTPD
 import fi.iki.elonen.NanoWSD
+import fi.iki.elonen.NanoWSD.WebSocketFrame.CloseCode
 import java.io.IOException
+import java.lang.Exception
 
 /**
  * An extension of [NanoWSD] for devtools to connect Arcs to a remote device for debugging.
  */
-class DevWebSocket : NanoWSD(12345) {
-    private var wsdSocket: WsdSocket? = null
+object DevWebServerImpl : DevWebServer, NanoWSD("localhost", 33317) {
+
+    private var wsdSockets = mutableListOf<WsdSocket>()
+    private var started = false
     private val log = TaggedLog { "DevWebSocket" }
 
     /**
      * Send a string to the client.
      */
-    fun send(msg: String) {
-        if (wsdSocket?.open ?: false) {
-            wsdSocket?.send(msg)
-        } else {
-            log.debug { "WebSocket Closed, can't send message [message=$msg]." }
+    override fun send(msg: String) {
+        wsdSockets.forEach { socket ->
+            socket.send(msg)
         }
     }
 
     override fun openWebSocket(ihttpSession: NanoHTTPD.IHTTPSession?): WebSocket {
-        wsdSocket = WsdSocket(ihttpSession, log)
-        return wsdSocket!!
+        val socket = WsdSocket(ihttpSession, log) { removeSocket(it) }
+        wsdSockets.add(socket)
+        return socket
+    }
+
+    fun startIfNeeded() {
+        try {
+            if (!started) {
+                super.start()
+                started = true
+            }
+        } catch (e: Exception) {
+            log.debug(e) { "Can't open Websocket. Error: [$e]." }
+        }
+    }
```

Comment:
Yeah I don't think we need this one, since we should only ever be calling `start` once in onCreate anyway.